### PR TITLE
Update to ASP.NET Core 2.0 and .NET Standard 2.0

### PR DESCRIPTION
**Pull request by scottkuhl**
I updated the library project to .NET Standard 2.0 and the extension to use the new ASP.NET Core 2.0 format.

To update the example, I just created a new ASP.NET Core 2.0 MVC project and added TurboLinks.NET.

I also changed the version number to 1.1.2, but I was not sure if that's the increment you want.

### CODE REVIEW

File: TurboLinkExample2/wwwroot/lib/jquery-validation/LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+The MIT License (MIT)
```
Comment (TerribleDev) [inline TurboLinkExample2/wwwroot/lib/jquery-validation/LICENSE.md:1]: I'm not going to allow libraries as commits

File: src/TurboLinks.Net/TurboLinks.Net.csproj [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <Description>Add support for Turbolinks</Description>
+    <VersionPrefix>1.1.1</VersionPrefix>
+    <Authors>Tommy Parnell</Authors>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (TerribleDev) [inline src/TurboLinks.Net/TurboLinks.Net.csproj:7]: I'm not going to drop support for netstandard1.6. Please target both. I am not going to introduce a breaking change such as this.

File: src/TurboLinks.Net/TurboLinks.Net.csproj [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    <AssemblyName>TurboLinks.Net</AssemblyName>
+    <PackageId>TurboLinks.Net</PackageId>
+    <PackageTags>TurboLinks Middlewear</PackageTags>
+    <PackageProjectUrl>https://github.com/tparnell8/TurboLinks.Net</PackageProjectUrl>
+    <PackageLicenseUrl>https://opensource.org/licenses/MIT</PackageLicenseUrl>
+    <NetStandardImplicitPackageVersion>1.6.0</NetStandardImplicitPackageVersion>
```
Comment (TerribleDev) [inline src/TurboLinks.Net/TurboLinks.Net.csproj:13]: I don't believe we need this.

File: src/TurboLinks.Net/TurboLinks.Net.csproj [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  <PropertyGroup>
+    <Description>Add support for Turbolinks</Description>
+    <VersionPrefix>1.1.1</VersionPrefix>
+    <Authors>Tommy Parnell</Authors>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <AssemblyName>TurboLinks.Net</AssemblyName>
+    <PackageId>TurboLinks.Net</PackageId>
+    <PackageTags>TurboLinks Middlewear</PackageTags>
+    <PackageProjectUrl>https://github.com/tparnell8/TurboLinks.Net</PackageProjectUrl>
```
Comment (TerribleDev) [inline src/TurboLinks.Net/TurboLinks.Net.csproj:11]: Please change this url to `https://github.com/terribledev/TurboLinks.Net`

File: src/TurboLinks.Net/TurboLinks.Net.csproj [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    <PackageLicenseUrl>https://opensource.org/licenses/MIT</PackageLicenseUrl>
+    <NetStandardImplicitPackageVersion>1.6.0</NetStandardImplicitPackageVersion>
+    <PackageTargetFallback>$(PackageTargetFallback);dnxcore50</PackageTargetFallback>
```
Comment (TerribleDev) [inline src/TurboLinks.Net/TurboLinks.Net.csproj:14]: I don't believe we need this

File: src/TurboLinks.Net/TurboLinks.Net.csproj [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    <GenerateAssemblyTitleAttribute>false</GenerateAssemblyTitleAttribute>
```
Comment (TerribleDev) [inline src/TurboLinks.Net/TurboLinks.Net.csproj:15]: Please adjust the AssemblyInfo, so these attributes can be generated.

File: src/TurboLinks.Net/TurboLinks.cs [hunk 7]
```diff
@@ -24,13 +24,13 @@ public TurboLinks(RequestDelegate next)
             var request = context.Request;
             var response = context.Response;
 
-            if(!string.IsNullOrWhiteSpace(request.Headers["X-XHR-Referer"]))
+            if (!string.IsNullOrWhiteSpace(request.Headers["X-XHR-Referer"]))
```
Comment (TerribleDev) [inline src/TurboLinks.Net/TurboLinks.cs:27]: Please remove the space, this is changing the style rules of the repo for no reason.

**Review by TerribleDev — CHANGES_REQUESTED** [review]
Thanks so much for your PR! I've left some changes in line.

I noticed you have made many modifications to the examples directory, but you have also made TurboLinkExample2. Can you explain to me why we need both?

Finally, the directory `TurboLinkExample2` does not follow the naming convention of `TurboLinks`, its pluralized in the repo. If we're to keep the directory, I would want it pluralized.

File: src/TurboLinks.Net/TurboLinks.Net.csproj [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    <GenerateAssemblyDescriptionAttribute>false</GenerateAssemblyDescriptionAttribute>
+    <GenerateAssemblyConfigurationAttribute>false</GenerateAssemblyConfigurationAttribute>
+    <GenerateAssemblyCompanyAttribute>false</GenerateAssemblyCompanyAttribute>
+    <GenerateAssemblyProductAttribute>false</GenerateAssemblyProductAttribute>
+    <GenerateAssemblyCopyrightAttribute>false</GenerateAssemblyCopyrightAttribute>
+  </PropertyGroup>
+
+  <ItemGroup>
```
Comment (TerribleDev) [inline src/TurboLinks.Net/TurboLinks.Net.csproj:23]: When you target both netstandard1.6, and 2.0. Please make this item group conditional on targetframework 1.6, and make a new ItemGroup with updated versions of these packages for netstandard2.0

**Comment by TerribleDev** [comment]
You also need to fix the build

**Comment by scottkuhl** [comment]
TurboLinkExample2 was a first attempt at updating the test project.  It should not have been included.  I'll pull it out.   I'll try to get all the changes in this weekend.

**Comment by scottkuhl** [comment]
I am going to close this PR and submit a new one with only the change to support ASP.NET Core 2.0.

**Pull request closed without merging**